src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -0,0 +1,155 @@
+import org.sagebionetworks.aws.AwsClientFactory;
+
+public class AppConfigServlet extends HttpServlet {
+
+  public AWSAppConfigData appConfigDataClient;
+  public Supplier<String> configSupplier;
+  public String configurationToken;
+  private StackConfiguration stackConfiguration;
+  private final String defaultConfigValue =
+    "{\"default configuration\":\"true\"}";
+
+  private String lastConfigValue = "";
+
+  private static final Logger logger = Logger.getLogger(
+    AppConfigServlet.class.getName()
+  );
+
+  @Inject
+  public AppConfigServlet(
+    AWSAppConfigData appConfigDataClient,
+    StackConfiguration stackConfiguration
+  ) {
+    this.appConfigDataClient = appConfigDataClient;
+    this.stackConfiguration = stackConfiguration;
+  }
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    initializeAppConfigClient();
+  }
+
+  public void initializeAppConfigClient() {
+    if (appConfigDataClient == null) {
+      appConfigDataClient = AwsClientFactory.createAppConfigClient();
+    }
+    startConfigurationSession();
+    initializeConfigSupplier();
+  }
+
+  public void startConfigurationSession() {
+    try {
+      StartConfigurationSessionRequest sessionRequest =
+        new StartConfigurationSessionRequest()
+          .withApplicationIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-AppConfigApp"
+          )
+          .withEnvironmentIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-environment"
+          )
+          .withConfigurationProfileIdentifier(
+            stackConfiguration.getStack() +
+            "-" +
+            stackConfiguration.getStackInstance() +
+            "-portal-configurations"
+          );
+      StartConfigurationSessionResult sessionResponse =
+        appConfigDataClient.startConfigurationSession(sessionRequest);
+      configurationToken = sessionResponse.getInitialConfigurationToken();
+    } catch (Exception e) {
+      configurationToken = null;
+    }
+  }
+
+  public void initializeConfigSupplier() {
+    configSupplier =
+      Suppliers.memoizeWithExpiration(
+        this::getLatestConfiguration,
+        5,
+        TimeUnit.MINUTES
+      );
+  }
+
+  public String getLatestConfiguration() {
+    if (configurationToken == null) {
+      return defaultConfigValue;
+    }
+    try {
+      GetLatestConfigurationRequest latestConfigRequest =
+        new GetLatestConfigurationRequest()
+          .withConfigurationToken(configurationToken);
+      GetLatestConfigurationResult latestConfigResponse =
+        appConfigDataClient.getLatestConfiguration(latestConfigRequest);
+      configurationToken = latestConfigResponse.getNextPollConfigurationToken();
+      ByteBuffer configData = latestConfigResponse.getConfiguration();
```

Comment:
From [the docs](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/appconfigdata/model/GetLatestConfigurationResult.html#getConfiguration--): 
> It is recommended to call ByteBuffer.duplicate() or ByteBuffer.asReadOnlyBuffer() before using or reading from the buffer. This behavior will be changed in a future major version of the SDK. 

```suggestion
      ByteBuffer configData = latestConfigResponse.getConfiguration().asReadOnlyBuffer();
```